docs/docs/how-tos/wait-user-input-functional.ipynb

```diff
@@ -300,35 +300,28 @@
     "Our tasks are otherwise unchanged from the [ReAct agent guide](../../how-tos/react-agent-from-scratch-functional):\n",
     "\n",
     "1. **Call model**: We want to query our chat model with a list of messages.\n",
-    "2. **Call tool**: If our model generates tool calls, we want to execute them.\n",
+    "2. **Call tool**: If our model generates tool calls, we want to execute them. **Note**: we're creating a new task for each tool by wrapping the `tool.invoke` in a [`task`][langgraph.func.task] function. Each tool task has the same name as the corresponding tool (e.g. `get_weather`)\n",
     "\n",
     "We just have one more tool accessible to the model."
    ]
   },
   {
    "cell_type": "code",
-   "execution_count": 9,
+   "execution_count": 10,
    "metadata": {},
    "outputs": [],
    "source": [
     "from langchain_core.messages import ToolMessage\n",
     "from langgraph.func import entrypoint, task\n",
     "\n",
-    "tools_by_name = {tool.name: tool for tool in tools}\n",
+    "tasks_by_name = {tool.name: task(tool.invoke, name=tool.name) for tool in tools}\n",
     "\n",
     "\n",
     "@task\n",
     "def call_model(messages):\n",
     "    \"\"\"Call model with a sequence of messages.\"\"\"\n",
     "    response = model.bind_tools(tools).invoke(messages)\n",
-    "    return response\n",
-    "\n",
-    "\n",
-    "@task\n",
-    "def call_tool(tool_call):\n",
```

Comment:
why can we not use call tool?

Reply:
we can! this was just an alternative approach